crates/sui-core/src/authority.rs

```diff
@@ -2808,6 +2815,23 @@ impl AuthorityState {
         }
     }
 
+    /// Acquire the execution lock while setting transaction locks during signing.
+    /// Required to ensure that we do not reconfigure while signing a transaction.
```

Comment:
Could you explain more in comments why we need this?

Reply:
thanks for asking - it actually isn't anymore, but i didn't notice. Originally, when we were storing locks in the perpetual db, they were stored with their epoch of origin, which was used for expiration. So, if there were two signing threads, one in the old epoch and one in the new, the new epoch thread could overwrite locks that the old epoch thread was trying to revert.

But this is unnecessary since I moved locks to the epoch db and removed the epoch from the lock.